fetch-builds-by-date-range-from-all-connected-controllers.groovy

```diff
@@ -0,0 +1,26 @@
+import com.cloudbees.opscenter.server.clusterops.steps.MasterGroovyClusterOpStep
+
+def retour = ''
+def stream = new ByteArrayOutputStream();
+def listener = new StreamBuildListener(stream);
+
+Jenkins.instance.getAllItems(com.cloudbees.opscenter.server.model.ConnectedMaster).each{
+    it.channel.call(new MasterGroovyClusterOpStep.Script("""
```

Comment:
You need to check that the controller is connected, with `if(it.channel) { ... }`. Otherwise you might hit an NPE on any stopped / disconnected controller.

Reply:
That's true, thanks! I have resolved this using the safe operator `?`  instead.